tests/test_rat_board.py

```diff
@@ -262,3 +262,29 @@ def test_next_free_mixed_board(self, keys: list, expected: int, rat_board_fx: Ra
 
         # and verify its functionality
         assert expected == nextFree
+
+    @pytest.mark.parametrize("test_input", [
+                            "UNIT_TEST",
+                            "@12345678-9876-53d1-ea5e-0000deadbeef",
+                            "42"])
+    def test_search_valid(self, test_input, rat_board_fx: RatBoard, rescue_plain_fx: Rescue):
+        test_rescue = rescue_plain_fx
+        test_rescue.uuid = UUID('12345678-9876-53d1-ea5e-0000deadbeef')
+        rat_board_fx.append(test_rescue)
+        assert rat_board_fx.search(test_input) == test_rescue
+
+    @pytest.mark.parametrize("test_input", [
+                            "unit_tes",
+                            '@12345678-9876-53d1-ea5e-0000dead',
+                            '@12345678-9876-53d1-ea5e-0000deadbee-',
+                            '@12345678-9876-53d1-ea5e-000deadsheep',
+                            42,
+                            "42.5"])
+    def test_search_garbage(self, test_input, rat_board_fx: RatBoard, rescue_plain_fx: Rescue):
+        test_rescue = rescue_plain_fx
+        test_rescue.uuid = UUID('12345678-9876-53d1-ea5e-0000deadbeef')
+        rat_board_fx.append(test_rescue)
+        try:
+            assert rat_board_fx.search(test_input) is None
+        except ValueError:
```

Comment:
```py
with pytest.raises(ValueError):
    assert rat_board_fx.search(test_input) is None

Reply:
This will require search() to raise an exception any time it can't find a case. Is this what we want?

Reply:
no? it would appear this test is asserting an exception was raised. if that is not the case remove the second unit.

Tests should only assert one thing.